### PR TITLE
DAOS-6745 Systemd: Make systemd unit files conditionally install base…

### PR DESCRIPTION
**Pull request by dpquigl**
…d on version

Some of the systemd unit file directives used change their name between the
versions of systemd. This means depending on which platform you build/install
DAOS on the unit files will be incorrect. Modify the rpm spec to selectively
install a pre systemd 230 unit file on EL7 and a post systemd 230 unit file on
everything else.

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -233,8 +233,14 @@ PREFIX="%{?_prefix}"
 mkdir -p %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/
 echo "%{_libdir}/daos_srv" > %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 mkdir -p %{?buildroot}/%{_unitdir}
+echo "rhel %{?rhel}"
+%if (0%{?rhel} <= 7)
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:237]: I think you want:

```suggestion
%if (0%{?rhel} >= 7)
```
here, to do the next two lines on EL7 (and higher).  But really, you probably need to assess the situation on EL8 and see if this is needed there and if not:
```suggestion
%if (0%{?rhel} = 7)
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: utils/rpms/daos.spec (hunk 1 above)
Comment (dpquigl) [reply utils/rpms/daos.spec]: Ok trying to do an exact match against EL7 and otherwise use the newer versions.

**Review by brianjmurrell — COMMENTED** [review]
Is it worth adding a test for invalid directives in unit files using `systemd-analyze verify foo.service` to detect this sort of (otherwise silent) issue in the future?  Such a test would serve to flag if EL7 does ever update to a newer systemd without us realizing.

**Comment by dpquigl** [comment]
@brianjmurrell Where would you suggest adding that check?

**Comment by brianjmurrell** [comment]
> @brianjmurrell Where would you suggest adding that check?

Adding as a small, basic functional test in `src/tests/ftest/<subdir>` (maybe `control`, maybe there is something else better) seems like it would provide the best coverage.  https://avocado-framework.readthedocs.io/en/62.0/WritingTests.html#turning-errors-into-failures probably provides a nice basic example of how to make a test out of running a process and turning it's result into an avocado result.

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -234,8 +234,13 @@ PREFIX="%{?_prefix}"
 mkdir -p %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/
 echo "%{_libdir}/daos_srv" > %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 mkdir -p %{?buildroot}/%{_unitdir}
+%if (0%{?rhel} == 7)
```
Comment (tanabarr) [inline utils/rpms/daos.spec:237]: why not < 8? we don't necessarily support earlier versions but it seems more intuitive

**Review by tanabarr — APPROVED** [review]

File: utils/rpms/daos.spec (hunk 2 above)
Comment (brianjmurrell) [reply utils/rpms/daos.spec:237]: Because systemd didn't exist on < 7.

Comment (tanabarr) [reply utils/rpms/daos.spec:237]: good point, thanks

**Comment by dpquigl** [comment]
@brianjmurrell We would like to merge this so we have fixed unit files for 1.2 and introduce the test later. Are you ok with that?

**Comment by brianjmurrell** [comment]
> @brianjmurrell We would like to merge this so we have fixed unit files for 1.2 and introduce the test later. Are you ok with that?

Sure.  It was just a suggestion to avoid this problem in the future.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**